src/hdmf/common/table.py

```diff
@@ -1192,6 +1192,17 @@ def to_dataframe(self, **kwargs):
         ret = self.__get_selection_as_df(sel)
         return ret
 
+    def generate_html_repr(self, level=0, access_code=""):
+        out = ""
+        for key, value in self.fields.items():
+            if key not in ("id", "colnames", "columns"):
+                out += self._generate_field_html(key, value, level, access_code)
+        out += (
+            f'<details><summary style="display: list-item; margin-left: {level * 20}px;" '
+            f'class="container-fields field-key" title="{access_code}"><b>table</b></summary>{self.to_dataframe().to_html()}</details>'
```

Comment:
My only concern here is that this will load the whole data into memory, and as such, rendering the tables may be slow, in particular if a file is opened remotely (e.g., on S3) or when dealing with large tables. I'm wondering whether we can have some form of configuration option to define: a) whether tables should be rendered, and b) when rendering tables, how many rows should be loaded.

Reply:
What if we do the following: If there are more than 4 rows, we slice the table to be the first four rows (returns a dataframe) and add text at the bottom of the repr saying how many rows there are? 

```
{self[0:min(4, len(self))].to_dataframe().to_html()}<br/>
{len(self)} rows x {len(self.columns)} columns
```

Reply:
this sounds like a good solution!